src/shared/calendarspec.c

```diff
@@ -369,14 +369,13 @@ int calendar_spec_to_string(const CalendarSpec *c, char **p) {
         }
 
         r = fflush_and_check(f);
+        fclose(f);
+
         if (r < 0) {
                 free(buf);
-                fclose(f);
```

Comment:
Strictly speaking, this was buggy, since `free` should be called *after* `fclose`. In practice, it was very unlikely to matter because of the flush above...